# Internship Program Analysis

## Problem Statement

Internship programs involve multiple departments, mentors, and participants, making it challenging to monitor enrollment, track intern progress, and evaluate overall program performance. Analyzing internship data helps identify completion trends, dropout patterns, department participation, internship duration, and mentor engagement. These insights support data-driven decision-making and help improve future internship programs.

## Project Objectives

The main objectives of this analysis are to:

- Analyze internship enrollment trends.
- Calculate internship completion and dropout rates.
- Examine department-wise participation.
- Analyze internship duration and completion progress.
- Evaluate mentor interactions.
- Generate meaningful visualizations and insights.

## Dataset Information

This dataset contains internship records for the year **2025**. It includes information about interns, educational background, departments, internship tracks, mentors, enrollment dates, internship progress, task submissions, mentor interactions, and completion status.

The dataset is used to perform exploratory data analysis (EDA) to identify patterns and generate insights related to internship program performance.

### Dataset Summary

| Attribute | Details |
|-----------|---------|
| **Dataset Name** | Internship Program Dataset |
| **Time Period** | 2025 |
| **File Format** | CSV (.csv) |
| **Number of Records** | 70,000 |
| **Number of Features** | 17 |
| **Data Type** | Structured Tabular Data |
| **Primary Goal** | Internship Program Analysis |

### Dataset Features

| Feature | Description |
|----------|-------------|
| **Intern_ID** | Unique identifier assigned to each intern |
| **Intern_Name** | Name of the intern |
| **Gender** | Gender of the intern |
| **Age** | Age of the intern |
| **City** | City of residence |
| **Education** | Educational qualification |
| **Institute** | Educational institute |
| **Department** | Internship department |
| **Internship_Track** | Internship specialization |
| **Mentor_ID** | Unique identifier assigned to each mentor |
| **Mentor_Name** | Name of the assigned mentor |
| **Enrollment_Date** | Internship enrollment date |
| **Internship_Duration_Weeks** | Total internship duration |
| **Weeks_Completed** | Number of weeks completed |
| **Tasks_Submitted** | Number of submitted tasks |
| **Mentor_Interactions** | Total mentor interactions |
| **Status** | Internship completion status |

## Import Required Libraries

This project uses Python libraries for data manipulation, analysis, and visualization. Pandas and NumPy are used for handling and processing data, while Matplotlib and Seaborn are used to create clear and informative visualizations that help identify patterns and trends within the dataset.

In [1]:
# Import Pandas for data manipulation and analysis
import pandas as pd

# Import NumPy for numerical computations
import numpy as np

# Import Matplotlib for creating visualizations
import matplotlib.pyplot as plt

# Import Seaborn for statistical data visualization
import seaborn as sns

## Load the Dataset

In this section, the internship dataset is loaded into a Pandas DataFrame. The `Enrollment_Date` column is converted to a datetime format to enable accurate date-based analysis and visualization throughout the project.

In [2]:
# Load the dataset
internship_df = pd.read_csv("../data/raw/internship_program_2025.csv")

# Convert the Enrollment_Date column to datetime format
internship_df["Enrollment_Date"] = pd.to_datetime(internship_df["Enrollment_Date"])

## Dataset Overview

This section provides an overview of the dataset to verify its structure, understand the available features, and ensure that the data has been loaded correctly before performing further analysis.

In [3]:
# Display the first five records
internship_df.head()

,Intern_ID,Intern_Name,Gender,Age,City,Education,Institute,Department,Internship_Track,Mentor_ID,Mentor_Name,Enrollment_Date,Internship_Duration_Weeks,Weeks_Completed,Tasks_Submitted,Mentor_Interactions,Status
0,INT00001,Ahmed Shafiq,Male,21,Peshawar,BS Computer Science,COMSATS University Islamabad,App Development,Android,MEN020,Huzaifa Waheed,2025-02-14,8,4,0,8,Dropped
1,INT00002,Adeel Chishti,Male,20,Rahim Yar Khan,FSC (Pre-Engineering),Government College Faisalabad,Frontend Development,React,MEN007,Fatima Ali,2025-08-25,8,8,8,17,Completed
2,INT00003,Areeba Hussain,Female,25,Multan,MS Computer Science,Government College University (GCU),Frontend Development,React,MEN007,Fatima Ali,2025-04-09,8,8,8,19,Completed
3,INT00004,Rafay Chaudhry,Male,24,Gwadar,BS Data Science,Riphah International University,Backend Development,PHP,MEN001,Ahmed Khan,2025-06-25,8,8,6,16,Completed
4,INT00005,Rafay Hashmi,Male,24,Gujranwala,BS Software Engineering,Riphah International University,Frontend Development,HTML CSS JS,MEN006,Ayesha Noor,2025-04-30,8,8,6,20,Completed


In [4]:
# Display dataset information
internship_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 70000 entries, 0 to 69999
Data columns (total 17 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Intern_ID                  70000 non-null  str           
 1   Intern_Name                70000 non-null  str           
 2   Gender                     70000 non-null  str           
 3   Age                        70000 non-null  int64         
 4   City                       70000 non-null  str           
 5   Education                  70000 non-null  str           
 6   Institute                  70000 non-null  str           
 7   Department                 70000 non-null  str           
 8   Internship_Track           70000 non-null  str           
 9   Mentor_ID                  70000 non-null  str           
 10  Mentor_Name                70000 non-null  str           
 11  Enrollment_Date            70000 non-null  datetime64[us]
 12  Internship_Dura

### Observation

- The dataset contains **70,000 records** and **17 features**.
- All columns contain complete records with no missing values.
- The dataset includes categorical, numerical, and datetime features.
- The `Enrollment_Date` column has been successfully converted to the datetime data type.
- The dataset is ready for data quality assessment and exploratory data analysis.

## Data Validation

Before starting the exploratory data analysis, the dataset is validated to ensure that it follows the expected business rules. This includes checking categorical values, identifying duplicate records, and verifying that numerical values fall within valid ranges.

In [5]:
# Check for duplicate records
print("Duplicated Records: ",internship_df.duplicated().sum())

Duplicated Records:  0


In [6]:
# Display unique values in important categorical columns
categorical_columns = [
    "Gender",
    "City",
    "Education",
    "Institute",
    "Department",
    "Internship_Track",
    "Status"
]

for column in categorical_columns:
    print("=" * 100)
    print(f"{column} ({internship_df[column].nunique()} Unique Values)")
    print("-" * 100)

    values = sorted(internship_df[column].unique())

    for value in values:
        print(f"• {value}")

    print()



Gender (2 Unique Values)
----------------------------------------------------------------------------------------------------
• Female
• Male

City (40 Unique Values)
----------------------------------------------------------------------------------------------------
• Abbottabad
• Bahawalpur
• Bannu
• Chiniot
• Dera Ghazi Khan
• Faisalabad
• Gujranwala
• Gujrat
• Gwadar
• Hyderabad
• Islamabad
• Jacobabad
• Jhang
• Karachi
• Kasur
• Khushab
• Kohat
• Lahore
• Larkana
• Mardan
• Mingora
• Mirpur Khas
• Multan
• Muzaffarabad
• Nawabshah
• Nowshera
• Okara
• Pakpattan
• Peshawar
• Quetta
• Rahim Yar Khan
• Rawalpindi
• Sahiwal
• Sargodha
• Sheikhupura
• Sialkot
• Sukkur
• Turbat
• Vehari
• Wah Cantt

Education (19 Unique Values)
----------------------------------------------------------------------------------------------------
• BBA
• BS Artificial Intelligence
• BS Computer Engineering
• BS Computer Science
• BS Cyber Security
• BS Data Analytics
• BS Data Science
• BS Information Tech

### Validate Business Rules

The following checks verify that the numerical values satisfy the expected business rules defined for the internship program.

In [7]:
# Validate internship duration

internship_df["Internship_Duration_Weeks"].value_counts()

Internship_Duration_Weeks
8    70000
Name: count, dtype: int64

In [8]:
# Validate completed weeks

print(f"Minimum Weeks Completed : {internship_df['Weeks_Completed'].min()}")

print(f"Maximum Weeks Completed : {internship_df['Weeks_Completed'].max()}")

Minimum Weeks Completed : 1
Maximum Weeks Completed : 8


In [9]:
# Validate submitted tasks

print(f"Minimum Tasks Submitted : {internship_df['Tasks_Submitted'].min()}")

print(f"Maximum Tasks Submitted : {internship_df['Tasks_Submitted'].max()}")

Minimum Tasks Submitted : 0
Maximum Tasks Submitted : 8


In [10]:
# Validate age values

print(f"Minimum Age : {internship_df['Age'].min()}")

print(f"Maximum Age : {internship_df['Age'].max()}")

Minimum Age : 17
Maximum Age : 32


In [11]:
# Validate mentor interactions

print(f"Minimum Mentor Interactions : {internship_df['Mentor_Interactions'].min()}")

print(f"Maximum Mentor Interactions : {internship_df['Mentor_Interactions'].max()}")

Minimum Mentor Interactions : 2
Maximum Mentor Interactions : 20


### Observation

- No duplicate records were found in the dataset.
- All categorical features contain valid and consistent categories.
- The numerical values satisfy the defined business rules.
- No invalid or unexpected values were identified during the validation process.
- The dataset is ready for exploratory data analysis.

## Exploratory Data Analysis (EDA)

Exploratory Data Analysis (EDA) is performed to understand the characteristics of the internship dataset and identify meaningful patterns through statistical analysis and data visualization. The analysis focuses on enrollment trends, completion status, department participation, internship progress, and mentor engagement.

### Monthly Internship Enrollment Trend

This analysis examines the monthly distribution of internship enrollments throughout 2025. Understanding enrollment trends helps identify peak registration periods and supports effective planning for recruitment, onboarding, mentor allocation, and training resources.

In [17]:
# Monthly Internship Enrollment Analysis

# Extract the enrollment month from the enrollment date
internship_df["Enrollment_Month"] = (
    internship_df["Enrollment_Date"].dt.month_name()
)

# Define the chronological order of enrollment months
month_order = [
    "January", "February", "March", "April",
    "May", "June", "July", "August",
    "September", "October"
]

# Calculate the number of interns enrolled each month
monthly_enrollment = (
    internship_df["Enrollment_Month"]
    .value_counts()
    .reindex(month_order)
)

# Display the monthly enrollment summary
monthly_enrollment

Enrollment_Month
January       3611
February      4926
March         6882
April        10576
May          12550
June         11197
July          7665
August        5478
September     4314
October       2801
Name: count, dtype: int64

#### Observation

Monthly internship enrollments were not evenly distributed throughout the year. Enrollment gradually increased from January (3,611) to a peak in May (12,550), followed by a steady decline through October (2,801). This indicates a clear seasonal enrollment pattern.

#### Business Insight

The concentration of enrollments between April and June suggests that the organization experiences its highest recruitment workload during these months. Planning additional onboarding sessions, assigning more mentors, and increasing training capacity during the peak period can help improve the intern experience and operational efficiency.